### DISCUSSION TITLE
Issue when running on Windows

### DISCUSSION THREAD

**Discussion by CallumVass** [keathley/wallaby#593] [Q&A] [answered]
Hi,

I wasn't sure to create a new issue or discussion so I've gone with discussion for now.

I'm trying to set up Wallaby as described in this page, which is using 0.28 of Wallaby: https://www.tddphoenix.com/setting-up/

I've followed all the instructions and triple checked to make sure my code is correct, however, when I run the first smoke test - I get the following error:

```
[error] GenServer Wallaby.Chrome.Chromedriver terminating
** (stop) :eacces
    erlang.erl:2217: :erlang.open_port({:spawn_executable, 'c:/Development/elixir/chatter/_build/test/lib/wallaby/priv/run_command.sh'}, [:binary, :stream, :use_stdio, :stderr_to_stdout, :exit_status, {:args, ["chromedriver", "--log-level=OFF", "--port=59814"]}])
    (wallaby 0.28.0) lib/wallaby/chrome/chromedriver/server.ex:96: Wallaby.Chrome.Chromedriver.Server.handle_continue/2
    (stdlib 3.12) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib 3.12) gen_server.erl:388: :gen_server.loop/7
    (stdlib 3.12) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start_chromedriver}
```

I have configured the chromedriver to be on my PATH variable, for example I can run `chromedriver` from command line to see its running successfully. It's also the same version as what my Chrome is.

Here is the result of running `elixir -v`:

```
elixir -v
Erlang/OTP 22 [erts-10.7] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1]

Elixir 1.11.2 (compiled with Erlang/OTP 21)
```

Has anyone else found issues when running it on Windows? Thanks

**Comment by mhanberg** [comment, marked as answer]
Thanks for asking your question as a discussion @CallumVass!

It looks like you're getting an `:acces` error when erlang attempts to start the chromedriver process. 

This leads me to think it's a permissions issue on the chromerdriver executable, so that would be where I suggest looking first.

**Reply by CallumVass** [reply]
Hi, 
Thanks for the reply. I've done some digging and it seems related to the fact it is a .sh file. Maybe Windows doesn't play so nicely with it and a .bat file might be needed?

Running this from iex works fine:
```
System.cmd("chromedriver", [], into: IO.stream(:stdio, :line))
```

However, running this doesn't (notice I don't even pass any arguments and just try to run the .sh file:
```
System.cmd("c:/Development/elixir/chatter/_build/test/lib/wallaby/priv/run_command.sh", [], into: IO.stream(:stdio, :line))
** (ErlangError) Erlang error: :eacces
    erlang.erl:2217: :erlang.open_port({:spawn_executable, 'c:/Development/elixir/chatter/_build/test/lib/wallaby/priv/run_command.sh'}, [:use_stdio, :exit_status, :binary, :hide, {:args, 
[]}])
```

Anyway, I'll close this as it's not related to Wallaby but seems to be a more general issue. Thanks again.

**Reply by mhanberg** [reply]
I think that Wallaby should be able to run on Windows out of the box. There is a chance we need to ship with a `.bat` file as well to support windows systems that don't include bash.

**Reply by CallumVass** [reply]
Oh that would be great - I'm surprised no one has seen this issue before unless no one is developing on Windows it seems? 

I found this guy was having the same issue as me: https://elixirforum.com/t/test-driven-development-with-phoenix-self-published-free/35154/13

**Reply by CallumVass** [reply]
I've configured sh to be in my PATH as I have Git for Windows installed, this now works from within iex:
```
System.cmd("sh", ["c:/Development/elixir/chatter/_build/test/lib/wallaby/priv/run_command.sh", "chromedriver"], into: IO.stream(:stdio, :line))   
PID: 1192
Starting ChromeDriver 89.0.4389.23 (61b08ee2c50024bab004e48d2b1b083cdbdac579-refs/branch-heads/4389@{#294}) on port 9515
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully
```

**Reply by mhanberg** [reply]
Hmm, interesting. I'm pretty sure that the script uses bash so I'm surprised to see it works using `sh`

**Reply by CallumVass** [reply]
I think the best solution could be to have a .bat file and then call that if the OS is windows rather than the .sh file.

**Reply by mhanberg** [reply]
I've created this issue to track this: https://github.com/elixir-wallaby/wallaby/issues/594
